### ISSUE TITLE
Contact form to dB (or contact form with db) plug-in

### ISSUE THREAD

**Issue by aledoy** [ProjectNami/projectnami#136]
Is there a plug-in that works for project nami? I have installed fast secure contact form and it's working great. I wanted a way to see online via the admin page, the entries submitted. I saw a plug-in names contact form DB, initially it wouldn't load properly (showing errors) as the create table had incorrect syntax in it (probably for mySQL). I ran the create table manually and fixed the issue. However, the plug-in is unable to display results as it's using a select query that is not working in SQL. I tried to quert the DB manually and saw the entrie being saved correctly.

**Comment by patrickebates** [comment]
Do you have any of the query errors from either the PHP error log or the translation log?

**Comment by aledoy** [comment]
Below was the script being executed that we had to correct…
 
CREATE TABLE wp_cf7dbplugin_submits ( submit_time DECIMAL(16,4) NOT NULL, form_name NVARCHAR(127) , field_name NVARCHAR(127) , field_value NVARCHAR(MAX) , field_order INTEGER, file LONGBLOB);
 
Below is the select query it is using that doesn't work:
Query:
SELECT `submit_time` AS 'Submitted',
 GROUP_CONCAT(if(`file` is null or length(`file`) = 0, null, `field_name`)) AS 'fields_with_file'
FROM `wp_cf7dbplugin_submits` 
WHERE `form_name` = 'Contact Form'  
GROUP BY `submit_time` 
ORDER BY `submit_time` DESC
LIMIT 0,100

Is there a contact form with db that you can suggest us to use? A readily available and working plugin?

**Comment by patrickebates** [comment]
Don't believe we've had any feedback about a working contact form.  We tend to only hear about the broken ones ;)

The problem is the GROUP_CONCAT function.  There is no translation for Azure SQL DB.  But if you are running on a local SQL Server, there is an option.

http://groupconcat.codeplex.com/

**Comment by aledoy** [comment]
Thanks for the quick response. The codeplex won't work as it only solves the group_concat. The query still has mySQL syntax (like the if, single quotes (my earlier post didn't show the quotes, i'm re-pasting the query used by the plug-in).

SELECT `submit_time` AS 'Submitted',
 GROUP_CONCAT(if(`file` is null or length(`file`) = 0, null, `field_name`)) AS 'fields_with_file'
FROM `wp_cf7dbplugin_submits` 
WHERE `form_name` = 'Contact Form'  
GROUP BY `submit_time` 
ORDER BY `submit_time` DESC
LIMIT 0,100

I have tried several contact DB plug-ins (or contact form plug-ins that includes logs) but don't work due to mySQL.

**Comment by patrickebates** [comment]
Yes, I did miss the IF statement.  That will need to be modified in the plugin, I just need to think about it first.  As for the LIMIT, the built-in translations will handle it already.
